internal/driver/webhtml.go

```diff
@@ -741,22 +744,147 @@ function viewer(baseUrl, nodes) {
 {{end}}
+</tbody>
 </table>
 </div>
 
 {{template "script" .}}
-<script>viewer({{.BaseURL}}, {{.Nodes}})</script>
+<script>
+function makeSortable() {
+  // Regular expression to extract leading number and a suffix.
+  // E.g., "1.3s" => 1.3, "s"
+  let numParseRe = new RegExp("^([\\-+]?[0-9]+([.][0-9]+)?)(.*)$")
+
+  // Unit conversions (for toUnit values produced by measurement.go)
+  const units = new Map([
+    ["B", 1.0],
+    ["kB", 1024.0],
+    ["MB", 1024.0 * 1024.0],
+    ["GB", 1024.0 * 1024.0 * 1024.0],
+    ["TB", 1024.0 * 1024.0 * 1024.0 * 1024.0],
+    ["PB", 1024.0 * 1024.0 * 1024.0 * 1024.0 * 1024.0],
+    ["ns", 1e-9],
+    ["us", 1e-6],
+    ["ms", 1e-3],
+    ["s", 1.0],
+    ["mins", 60.0],
+    ["hrs", 3600.0],
+    ["days", 3600.0 * 24.0],
+    ["wks", 3600.0 * 24.0 * 7.0],
+    ["yrs", 3600.0 * 24.0 * 7.0 * 365.0],  // Ignore leap years like measurement.go
+  ])
+
+  // Which column are we currently sorted by and in what order?
+  let currentColumn = 0
+  let descending = true
+
+  function sortBy(column, numeric) {
+    const rows = document.getElementById("rows")
+    if (rows == null) return
+
+    // Update sort criteria
+    if (column == currentColumn) {
+      descending = !descending  // Reverse order
+    } else {
+      currentColumn = column
+      descending = (column != 5)  // Name is ascending by default
+    }
+
+    // Get info for all rows we want to sort.
+    let list = []
+    for (const r of rows.children) {
+      if (r.children.length <= column) continue
+      const c = r.children[column]
+      const val = c.innerText
+
+      // Extract numeric and string portions from val.
+      let num = 0
+      let suffix = val
+      if (numeric) {
+        const match = val.match(numParseRe)
+	if (match != null) {
+          const n = parseFloat(match[1])
+          if (!isNaN(n)) {
+            num = n
+            suffix = match[3]
+            if (units.has(suffix)) {
+              num *= units.get(suffix)
+              suffix = ""
+            }
+          }
+        }
+      }
+      list.push([num, suffix, r])
+    }
+
+    function cmp(a, b) {
+      if (a[0] < b[0]) return -1
+      if (a[0] > b[0]) return +1
+      if (a[1] < b[1]) return -1
+      if (a[1] > b[1]) return +1
+      return 0
+    }
+    function dcmp(a, b) { return -cmp(a, b) }
+
+    list.sort(descending ? dcmp : cmp)
+
+    // Sum% column holds the running sum of rows and since we
+    // have re-ordered rows, recompute that.
+    let sumPercent = 0.0
+    for (const r of list) {
+      const row = r[2]
+      if (row.children.length < 3) continue
+      const flatPercent = parseFloat(row.children[1].innerText)
+      if (!isNaN(flatPercent)) sumPercent += flatPercent
+      row.children[2].innerText = sumPercent.toFixed(2) + "%"
+    }
+
+    for (const r of list) {
+      rows.appendChild(r[2])
+    }
+  }
```

Comment:
It looks like a lot of work can be saved if Go generated just a table data for JS and JS always sorted those and then rendered new rows (for efficiency into a html fragment that can be then added into the tbody.rows. Something like (have not tried, just winging it here):

```
var tableData = [
  // generated by {{range ...}} as before only producing JS instead of HTML
  [{label: "2ms", value: 0.002}, {label: "30%", value: 30}, {label: "ny-name", value: "my-name"}],
  ...
];
...
function sortBy(column) {
  // select the cmp to be asc or desc, but will be simpler only doing a<b, a>b,
  tableData.sort((a, b) => cmp(a[column].value, b[column].value))
  const fragment = document.createDocumentFragment()
  for (const row of tableData) {
    const tr = document.createElement('tr')
    for (const cell of row) {
      const td = document.createElement('td')
      td.textContent = cell.label
      tr.appendChild(td)
    }
    fragment.appendChild(tr)
  }
  rows.textContent = "" // empty the table body first
  rows.appendChild(fragment)
}
```

Reply:
This is an interesting point.  It will require a fair bit of surgery, both here as well as in the Go code, but we might end up in an overall better place.  I'll try that out and get back to you.

Reply:
Done.